core/domain/prod_validation_jobs_one_off.py

```diff
@@ -648,13 +747,38 @@ def _validate_explorations_are_public(cls, item):
                     item.id, private_exp_ids))
 
     @classmethod
-    def _validate_collections_are_public(cls, item):
+    def _validate_collections_are_public(
+            cls, item, field_name_to_external_model_references):
         """Validates that collections for model are public.
 
         Args:
             item: ndb.Model. BaseUserModel to validate.
+            field_name_to_external_model_references:
+                dict(str, (list(ExternalModelReference))).
+                A dict is keyed by field name. Each value consists of a list
+                of ExternalModelReference objects.
         """
-        col_ids = cls._get_col_ids(item)
+        if 'collection_ids' not in field_name_to_external_model_references:
+            return
+
+        col_ids = []
+        collection_model_references = (
+            field_name_to_external_model_references['collection_ids'])
+
+        for collection_model_reference in collection_model_references:
+            model_class = collection_model_reference.class_name
+            model_id = collection_model_reference.model_id
+            collection_model = collection_model_reference.model_instance
+
+            if collection_model is None or collection_model.deleted:
+                cls.errors['collection_ids field check'].append((
```

Comment:
Here and elsewhere, probably better to do cls._add_error(..., ...) rather than reference cls.errors directly. Easier to read.

If the errors are of a common type then you can also probably pass in model_id and model_class as arguments.

Basically, see if you can find a way to reduce some of the duplication of the logic/interpolation so that just the common function is the only thing that's a bit complex, and all the per-class functions are really easy to parse.

Reply:
Added the function but I don't think it is possible to pass in arguments since all errors are formatted differently based on what the error message is related to.